jarviscli/plugins/news.py

```diff
@@ -58,22 +60,71 @@ def run(self, jarvis, s):
                 jarvis.say("I couldn't find news", Fore.RED)
         else:
             try:
+                if len(jarvis.get_data('news-settings')) > 0:
+                    self.is_news_site_configured = True
                 self.news(jarvis)
             except:
                 jarvis.say("I couldn't find news", Fore.RED)
 
+    @staticmethod
+    def get_key():
+        return "news-settings"
+
     def news(self, jarvis):
-        self.news_options(jarvis)
-        self.get_news()
+        print('Would you like to configure your own news channel(y/n)')
+        i = str(input())
+        if i.lower() == "y":
+            self.configure_news_site(jarvis)
+        else:
+            self.news_options(jarvis)
+            self.get_news()
 
     '''
         This is the quickest way to get news it also has the
         least amount of options for the user.
     '''
 
+    def configure_news_site(self, jarvis):
+        news_site_list = jarvis.get_data(News.get_key())
+        # Check if no sites are configured
+        if news_site_list is None:
+            news_site_list = []
+        print("List of configured news sites are as below")
+        for idx, site in enumerate(news_site_list):
+            print("{} : {}".format(idx + 1, site))
+
+        print("Search www.newsapi.org for the list of news channels. Enter site-key")
+        site_key = str(input())
+        if site_key is '':
+            print("Wrong input.")
+        elif site_key in news_site_list:
+            print("Site already added. Hence not updating the list.")
+        else:
+            news_site_list.append(site_key)
+            self.is_news_site_configured = True
+
+        print("List of configured news sites are as below")
+
+        for idx, site in enumerate(news_site_list):
+            print("{} : {}".format(idx + 1, site))
+
+        print("Would you like to delete any in the list?(y/n)")
+        del_from_list = str(input())
+        if del_from_list.lower() == "y":
+            print("Enter the site index to be deleted. If multiple sites, then enter index with comma seperated values")
+            index_to_del = str(input())
+            items_idx_to_del = index_to_del.split(',')
+            list_to_del = [news_site_list[int(item) - 1] for item in items_idx_to_del]
+            news_site_list = [item for item in news_site_list if item not in list_to_del]
+            if len(news_site_list) == 0:
+                self.is_news_site_configured = False
+        jarvis.update_data(News.get_key(), news_site_list)
+        jarvis.save()
```

Comment:
jarvis.save() does not exist... Memory save automatically after each add /  update / del - so no need to call save manually.

Reply:
Removed.